commander3/src/comm_tod_WMAP_mod.f90

```diff
@@ -19,469 +19,1237 @@
+      end if
+      call wall_time(t2); t_tot(10) = t_tot(10) + t2 - t1
+      call wall_time(t6)
+      if (self%myid == self%numprocs/2 .and. self%verbosity > 0) then
+         write(*,*) '  Time dist sky        = ', nint(t_tot(9))
+         write(*,*) '  Time sl precomp      = ', nint(t_tot(13))
+         write(*,*) '  Time decompress      = ', nint(t_tot(11))
+         write(*,*) '  Time project         = ', nint(t_tot(1))
+         write(*,*) '  Time orbital         = ', nint(t_tot(2))
+         write(*,*) '  Time sl interp       = ', nint(t_tot(12))
+         write(*,*) '  Time ncorr           = ', nint(t_tot(3))
+         write(*,*) '  Time gain            = ', nint(t_tot(4))
+         write(*,*) '  Time absgain         = ', nint(t_tot(14))
+         write(*,*) '  Time sel data        = ', nint(t_tot(15))
+         write(*,*) '  Time clean           = ', nint(t_tot(5))
+         write(*,*) '  Time noise           = ', nint(t_tot(6))
+         write(*,*) '  Time samp abs        = ', nint(t_tot(16))
+         write(*,*) '  Time samp bp         = ', nint(t_tot(17))
+         write(*,*) '  Time chisq           = ', nint(t_tot(7))
+         write(*,*) '  Time bin             = ', nint(t_tot(8))
+         write(*,*) '  Time scanlist        = ', nint(t_tot(20))
+         write(*,*) '  Time final           = ', nint(t_tot(10))
+         write(*,*) '  Time solving cg      = ', nint(t_tot(21))
+         write(*,*) '  Time per cg iter     = ', nint(t_tot(21)/num_cg_iters)
+         write(*,*) '  Number of cg iters   = ', num_cg_iters
+         write(*,*) '  Time writing to txt  = ', nint(t_tot(22))
+         write(*,*) '  Time total           = ', int(t6-t5), int(sum(t_tot(1:18)))
+      end if
+
+      ! Clean up temporary arrays
+      deallocate(A_abscal, b_abscal, chisq_S, procmask)
+      deallocate(b_map, M_diag, cg_tot)
+      if (allocated(b_mono)) deallocate (b_mono)
+      if (allocated(sys_mono)) deallocate (sys_mono)
+      if (allocated(slist)) deallocate (slist)
+      if (allocated(dipole_mod)) deallocate (dipole_mod)
+
+      if (allocated(outmaps)) then
+         do i = 1, nout
+            call outmaps(i)%p%dealloc
+         end do
+         deallocate (outmaps)
+      end if
+
+      deallocate (map_sky, cg_sol)
+      if (self%myid == 0) deallocate (r, rhat, s, r0, q, shat, p, phat, v, m_buf, determ)
+
+      if (correct_sl) then
+         do i = 1, self%ndet
+            call self%slconv(i)%p%dealloc(); deallocate(self%slconv(i)%p)
+         end do
+      end if
+
+      call int2string(iter, ctext)
+      call update_status(status, "tod_end"//ctext)
+
+      ! Parameter to check if this is first time routine has been
+      self%first_call = .false.
+
+   end subroutine process_WMAP_tod
+
+
+
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   ! Subroutine to save time-ordered-data chunk
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   subroutine write_tod_chunk(filename, tod)
```

Comment:
this feels like it could be in comm_tod_mod, it's not WMAP specific

Reply:
can do!